### PR TITLE
ENH: Add license checks

### PR DESCRIPTION
**Pull request by ericdill**
This adds checks to avoid mirroring agpl or unlicensed conda packages
internally.  cc @parente @mariusvniekerk

### CODE REVIEW

File: conda_mirror/conda_mirror.py [hunk 1]
```diff
@@ -73,6 +73,36 @@ def cli():
     main(args.upstream_channel, args.target_directory, args.platform)
 
 
+def not_in_upstream(local_repo_metadata, upstream_repo_metadata):
+    upstream_package_names = list(upstream_repo_metadata.keys())
+    local_package_names = list(local_repo_metadata.keys())
+    for pkg in upstream_package_names:
+        if pkg not in local_package_names:
```
Comment (mariusvniekerk) [inline conda_mirror/conda_mirror.py]: Repeated linear search here, this could get ugly for the initial mirror

Comment (ericdill) [inline conda_mirror/conda_mirror.py]: fair point

Comment (ericdill) [inline conda_mirror/conda_mirror.py]: updated

**Comment by mariusvniekerk** [comment]
LGTM

**Pull request merged**